packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -3265,38 +3273,54 @@ export default class Meeting extends StatelessWebexPlugin {
 
           if (this.config.stats.enableStatsAnalyzer) {
             // TODO: ** Dont re create StatsAnalyzer on reconnect or rejoin
-            this.statsAnalyzer = new StatsAnalyzer(this.config.stats);
+            this.networkQualityMonitor = new NetworkQualityMonitor(this.config.stats);
+            this.statsAnalyzer = new StatsAnalyzer(this.config.stats, this.networkQualityMonitor);
             this.statsAnalyzer.on(EVENT_TRIGGERS.MEDIA_QUALITY, (options) => {
               // TODO:  might have to send the same event to the developer
               // Add ip address info if geo hint is present
               options.data.intervalMetadata.peerReflexiveIP = this.webex.meetings.geoHintInfo?.clientAddress || options.data.intervalMetadata.peerReflexiveIP || MQA_STATS.DEFAULT_IP;
               Metrics.postEvent({event: eventType.MEDIA_QUALITY, meeting: this, data: {intervalData: options.data, networkType: options.networkType}});
             });
-            this.statsAnalyzer.on(EVENT_TRIGGERS.MEETING_HIGH_PACKETLOSS, (res) => {
-              if (res.mediaType === VIDEO) {
-                if (this.config.stats.autoDowngradeEnabled && this.canUpdateMedia()) {
-                  LoggerProxy.logger.log(`${LOG_HEADER} High Packet Loss Detected, Adjusting Local Video Quality`);
-                  if (this.mediaProperties.localQualityLevel === QUALITY_LEVELS.HIGH) {
-                    this.setLocalVideoQuality(QUALITY_LEVELS.MEDIUM);
-                  }
-                  else if (this.mediaProperties.localQualityLevel === QUALITY_LEVELS.MEDIUM) {
-                    this.setLocalVideoQuality(QUALITY_LEVELS.LOW);
-                  }
+            this.networkQualityMonitor.on(EVENT_TRIGGERS.NETWORK_QUALITY_JUDGMENT, (res) => {
+              Trigger.trigger(
+                this,
+                {
+                  file: 'meeting/index',
+                  function: 'addMedia'
+                },
+                EVENT_TRIGGERS.NETWORK_QUALITY_JUDGMENT,
+                {
+                  networkAcceptable: res.networkAcceptable,
+                  mediaType: res.mediaType,
+                  data: res.data
```

Comment:
how does the data look here ?

Reply:
networkAcceptable is a boolean - same as adequateExistingNetworkQuality
mediaType is same as - mediaType in networkQualityMonitor
data - is the networkQualityDataStructure object which includes a boolean for acceptable and the actual value from the stats object

it's in the documentation for networkQualityMonitor
do you want me to add it to the meeting/index.js file as well
/**
  * Meeting - network quality judgment event
  * Emitted on each interval of retrieving stats Analyzer data
  * @event network:quality:judgment
  * @type {Object}
  * @property {string} mediaType {video|audio}
  * @property {boolean} networkAcceptable is network overal acceptable
  * @property {Object} networkQualityDataStructure for each indicator type
  * @property {boolean} networkQualityDataStructure.acceptable
  * @property {Number} networkQualityDataStructure.value - score or RTC value for each indicator type
  * @memberof NetworkQualityMonitor
  */
/**